qemu/qemu.go

```diff
@@ -247,6 +249,12 @@ type Options struct {
 	// Extra kernel command-line arguments.
 	KernelArgs string
 
+	// OVMF_VARS.fd firmware location
+	OvmfVars string
+
+	// OVMF_CODE.fd firmware location
+	OvmfCode string
```

Comment:
I'm trying to make the qemu package the holder of the basic concepts, while other packages like firmware hold details like this. I think otherwise this is going to grow without bound.

IMO I'd do this:

```go
func WithDefaultUEFI() qemu.Fn {
  return WithUEFI("", "")
}

// WithUEFI sets the QEMU arguments for enabling UEFI with OVMF firmware.
//
// ovmfCode and ovmfVars are substituted by VMTEST_OVMF_CODE and VMTEST_OVMF_VARS if empty.
func WithUEFI(ovmfCode, ovmfVars string) qemu.Fn {
        if ovmfCode == "" {
                ovmfCode = os.Getenv("VMTEST_OVMF_CODE")
        }
        if ovmfVars == "" {
                ovmfVars = os.Getenv("VMTEST_OVMF_VARS")
        }
	return func(alloc *qemu.IDAllocator, opts *qemu.Options) error {
		opts.AppendQEMU(
			"-drive", fmt.Sprintf("if=pflash,format=raw,unit=0,file=%s,readonly=on", ovmfCode),
			"-drive", fmt.Sprintf("if=pflash,format=raw,unit=1,file=%s", ovmfVars),
		)
		return nil
	}
}
```

Reply:
I like this solution. Shame Go doesn't support function overloading :)